mapsforge-map/src/main/java/org/mapsforge/map/view/FrameBufferHA2.java

```diff
@@ -20,92 +20,102 @@
 import org.mapsforge.core.graphics.Bitmap;
 import org.mapsforge.core.graphics.GraphicContext;
 import org.mapsforge.core.graphics.GraphicFactory;
+import org.mapsforge.core.graphics.Matrix;
 import org.mapsforge.core.model.Dimension;
 import org.mapsforge.core.model.MapPosition;
+import org.mapsforge.core.model.Point;
 import org.mapsforge.map.model.DisplayModel;
 import org.mapsforge.map.model.FrameBufferModel;
 
 public class FrameBufferHA2 extends FrameBuffer {
 
     /*
      *  lm: layer manager
-     *      Layer manager draws the bitmap off-screen
+            Layer manager draws the bitmap off-screen
      *  od: onDraw() -> draw()
      *      swaps the two bitmaps and puts one bitmap to the screen
      *      while the layer manager draws the next off-screen bitmap.
+     *
      */
-    private final FrameBufferBitmap lmBitmap = new FrameBufferBitmap();
-    private final FrameBufferBitmap odBitmap = new FrameBufferBitmap();
 
-    private final Object dimLock = new Object();
+    private final FrameBufferBitmap odBitmap = new FrameBufferBitmap();
+    private final FrameBufferBitmap lmBitmap = new FrameBufferBitmap();
     private MapPosition lmMapPosition;
 
+    private final FrameBufferBitmap.Lock allowSwap = new FrameBufferBitmap.Lock();
+
+    private Dimension dimension;
+    private final Matrix matrix;
+
+    private final DisplayModel displayModel;
+    private final FrameBufferModel frameBufferModel;
+    private final GraphicFactory graphicFactory;
```

Comment:
Why not using `FrameBuffer` inherited variables and methods to avoid duplication?